rasa/shared/core/domain.py

```diff
@@ -569,34 +574,55 @@ def _collect_overridden_default_intents(
         }
         return sorted(intent_names & set(rasa.shared.core.constants.DEFAULT_INTENTS))
 
-    def _initialize_forms(self, forms: Union[Dict[Text, Any], List[Text]]) -> None:
-        """Initialize the domain's `self.form` and `self.form_names` attributes.
+    @staticmethod
+    def _initialize_forms(
+        forms: Union[Dict[Text, Any], List[Text]]
+    ) -> Tuple[List[Text], Dict[Text, Any], List[Text]]:
+        """Retrieves the initial values for the Domain's form fields.
 
         Args:
             forms: Form names (if forms are a list) or a form dictionary. Forms
                 provided in dictionary format have the form names as keys, and either
                 empty dictionaries as values, or objects containing
                 `SlotMapping`s.
+
+        Returns:
+            The form names, a mapping of form names and slot mappings, and custom
+            actions.
+            Returning custom actions for each forms means that Rasa Open Source should
+            not use the default `FormAction` for the forms, but rather a custom action
+            for it. This can e.g. be used to run the deprecated Rasa Open Source 1
+            `FormAction` which is implemented in the Rasa SDK.
         """
-        if not forms:
-            # empty dict or empty list
-            return
-        elif isinstance(forms, dict):
+        if isinstance(forms, dict):
             # dict with slot mappings
-            self.forms = forms
-            self.form_names = list(forms.keys())
-        elif isinstance(forms, list) and isinstance(forms[0], str):
-            # list of form names
-            self.forms = {form_name: {} for form_name in forms}
-            self.form_names = forms
-        else:
+            return list(forms.keys()), forms, []
+
+        if isinstance(forms, list) and (not forms or isinstance(forms[0], str)):
+            # list of form names (Rasa Open Source 1 format)
             rasa.shared.utils.io.raise_warning(
-                f"The `forms` section in the domain needs to contain a dictionary. "
-                f"Instead found an object of type '{type(forms)}'.",
+                "The `forms` section in the domain used the old Rasa Open Source 1 "
+                "list format to define forms. Rasa Open Source will be configured to "
+                "use the deprecated `FormAction` within the Rasa SDK. If you want to "
+                "use the new Rasa Open Source 2 `FormAction` adapt your `forms` "
+                "section as described in the documentation. Support for the "
+                "deprecated `FormAction` in the Rasa SDK will be removed in Rasa Open "
+                "Source 3.0.",
```

Comment:
```suggestion
                "The `forms` section in the domain used the old Rasa Open Source 1.0 "
                "list format to define forms. Rasa Open Source will be configured to "
                "use the deprecated `FormAction` within the Rasa SDK. If you want to "
                "use the new Rasa Open Source 2.0 `FormAction` adapt your `forms` "
                "section as described in the documentation. Support for the "
                "deprecated `FormAction` in the Rasa SDK will be removed in Rasa Open "
                "Source 3.0.",
```

Reply:
why do we need the `.0`? It was not only part of Rasa Open Source 1.0, but of all Rasa Open Source 1.x versions, right? Same for `2.0`: It's part of `2.x`